lens-follower-tracker/src/lib.rs

```diff
@@ -51,52 +66,65 @@ fn map_lens_follower_events(params: String, blk: eth::Block) -> Result<FollowEve
-                follow_nft_transferred_event.follower_address = to.clone();
-                follow_nft_transferred_event.follow_type =  FOLLOW.to_string();    
-            }
 
-            // scenario 2: follow for event [from:0x..782 [proxy] to: follower address]
-            // example tx hash: 0xbc3c64e31c901f0560fb55e1ff8635d91b5a7ab8f5d5b5d50b6053a3907b1f10
-            if from == proxy_action_contract_address && to != ZERO_ADDRESS {
-                follow_nft_transferred_event.follower_address = to.clone();
-                follow_nft_transferred_event.follow_type =  FOLLOW.to_string();  
-            }
+            profile_follow_nft_transferred_events_map.insert(key, follow_nft_transferred_event);
 
-            // scenario 3: burn event [from:follower to: 0x]
-            // example tx hash: 0x68d80b53990cc9b3d7e48d6c11295f2ede7cbff8ae6a05b7c265bcccf9d994bc
-            if from != proxy_action_contract_address && to == ZERO_ADDRESS {
-                follow_nft_transferred_event.follower_address = from.clone();
-                follow_nft_transferred_event.follow_type =  UNFOLLOW.to_string();               
+            continue;
+        }
+         // for Followed event
+         if let Some(event) = abis::lens_events::events::Followed::match_and_decode(log) {
+            for profile_id in event.profile_ids {
+                let key = format!("{}", profile_id);
+                log::info!("tx_hash: {} , profile_id {} has follow event", tx_hash, key);
+                profile_followed_event_exists_map.insert(key, true);
             }
+        } 
+    }
 
-            // scenario 4: transfer between addresses  -> should unfollow the from address
-            // example tx hash: 0xbac904222e15b792acd053f86a8834ebbb4414bedbf4051b0297c378bde73f38
-            if from != ZERO_ADDRESS && from != proxy_action_contract_address 
-                && to != ZERO_ADDRESS && to != proxy_action_contract_address {
-                    follow_nft_transferred_event.follower_address = from.clone();
-                    follow_nft_transferred_event.follow_type =  UNFOLLOW.to_string();                    
-            }
-            // finally push the data to vector
-            if follow_nft_transferred_event.follow_type != EMPTY_STRING {
-                follow_nft_transferred_events.push(follow_nft_transferred_event); 
-            }
+    // checks to update the correct follower address
+    for (key, value)  in  profile_followers_map.iter() {
+        let follower_address = value.clone();
+        let mut follow_nft_transferred_event = profile_follow_nft_transferred_events_map.get(key).unwrap().clone();
+        let is_followed_event_exists = *profile_followed_event_exists_map.get(&follow_nft_transferred_event.follow_profile_id).unwrap_or(&false);
+            // for actual follow events, we will have below is_followed_event_exists as true
+        if is_followed_event_exists {
```

Comment:
What happen if there is follow & unfollow action on same profile ID in a block. This logic will fail?

Reply:
is seems wrong that if any follow Event exists in block for that profile, we are considering all follow_type as FOLLOW